server/src/graql/gremlin/fragment/ValueFragment.java

```diff
@@ -147,19 +149,21 @@ public double estimatedCostAsStartingPoint(TransactionOLTP tx) {
         // this is probably not the highest quality heuristic (plus it is a heavy operation), needs work
 
         Label attributeLabel = Label.of("attribute");
+        long totalImplicitRels = 0;
+        long totalAttributes = 0;
 
         AttributeType attributeType = tx.getSchemaConcept(attributeLabel).asAttributeType();
         Stream<AttributeType> attributeSubs = attributeType.subs();
 
-        Label implicitAttributeRelation = Schema.ImplicitType.HAS.getLabel(attributeLabel);
-        SchemaConcept implicitAttributeRelationType = tx.getSchemaConcept(implicitAttributeRelation);
-        double totalImplicitRels = 0.0;
-        if (implicitAttributeRelationType != null) {
-            Stream<RelationType> implicitSubs = implicitAttributeRelationType.asRelationType().subs();
-            totalImplicitRels = implicitSubs.map(t -> statistics.count(tx, t.label().toString())).reduce((a,b) -> a+b).orElse(1L);
+        for (Iterator<AttributeType> it = attributeSubs.iterator(); it.hasNext(); ) {
+            AttributeType attrType = it.next();
+            Label attrLabel = attrType.label();
+            Label implicitAttrRelLabel = Schema.ImplicitType.HAS.getLabel(attrLabel);
```

Comment:
I think we might want to look at the `KEY` variant as well for implicit relations.

Reply:
actually, since it is an estimate, it might be ok